backend/src/main/java/com/stock/admin/model/entity/Shop.java

```diff
@@ -0,0 +1,177 @@
+package com.stock.admin.model.entity;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.index.Indexed;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.Date;
+import java.util.Objects;
+
+/**
+ * The type Shop.
+ */
+@Document("shops")
+public class Shop {
+    @Id
+    private String id;
+
+    private String code; //yet to decide on the data type of it
+
+    @Indexed(name = "shopName")
+    private String name;
+    private String location;
+    private Date createdAt;
```

Comment:
check whether `createdat` and `updatedat` can be epoch time i.e long datatype.. if that's easy we can use that.